src/webgpu/api/validation/encoding/cmds/compute_pass.spec.ts

```diff
@@ -0,0 +1,135 @@
+      - 'valid' buffers do not generate an error
+    - Tests that, for a buffer with 6 elements, indirect offsets:
+      - 0, 'sizeof(uint32)', and '3 * sizeof(uint32)' do not generate an error
+      - An error should be generate by the following:
+        - 1: non-multiple of 4
+        - '4 * sizeof(uint32): z-components outside of buffer
+`;
+
+import { params, poptions } from '../../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+
+import { ValidationTest } from './../../validation_test.js';
+
+class F extends ValidationTest {
+  createComputePipeline(state: 'valid' | 'invalid'): GPUComputePipeline {
+    if (state === 'valid') {
+      return this.createNoOpComputePipeline();
+    }
+
+    this.device.pushErrorScope('validation');
+    const pipeline = this.device.createComputePipeline({
+      computeStage: {
+        module: this.device.createShaderModule({
+          code: '',
+        }),
+        entryPoint: '',
+      },
+    });
+    this.device.popErrorScope();
+
+    return pipeline;
+  }
+
+  createIndirectBuffer(state: 'valid' | 'invalid' | 'destroyed', data: Uint32Array): GPUBuffer {
+    const descriptor: GPUBufferDescriptor = {
+      size: data.byteLength,
+      usage: GPUBufferUsage.INDIRECT | GPUBufferUsage.COPY_DST,
+    };
+
+    if (state === 'invalid') {
+      descriptor.usage = 0xffff; // Invalid GPUBufferUsage
+    }
+
+    this.device.pushErrorScope('validation');
+    const buffer = this.device.createBuffer(descriptor);
+    this.device.popErrorScope();
+
+    if (state === 'valid') {
+      this.queue.writeBuffer(buffer, 0, data);
+    }
+
+    if (state === 'destroyed') {
+      buffer.destroy();
+    }
+
+    return buffer;
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('set_pipeline')
+  .params(poptions('state', ['valid', 'invalid'] as const))
+  .fn(t => {
+    const pipeline = t.createComputePipeline(t.params.state);
+    const { encoder, finish } = t.createEncoder('compute pass');
+    encoder.setPipeline(pipeline);
+    t.expectValidationError(() => {
+      finish();
+    }, t.params.state === 'invalid');
+  });
+
+g.test('dispatch_sizes')
+  .params(
+    params()
+      .combine(poptions('dispatchType', ['direct', 'indirect'] as const))
+      .combine(
+        poptions('workSizes', [
+          [0, 0, 0],
+          [1, 1, 1],
+          // TODO: Add a test for a large workSize when the upper limit has been decided.
+        ] as [number, number, number][])
+      )
+  )
+  .fn(t => {
+    const pipeline = t.createNoOpComputePipeline();
+    const [x, y, z] = t.params.workSizes;
+    const { encoder, finish } = t.createEncoder('compute pass');
+    encoder.setPipeline(pipeline);
+    if (t.params.dispatchType === 'direct') {
+      encoder.dispatch(x, y, z);
+    } else if (t.params.dispatchType === 'indirect') {
+      encoder.dispatchIndirect(t.createIndirectBuffer('valid', new Uint32Array([x, y, z])), 0);
```

Comment:
note: I think a valid thing to do for invalid `dispatchIndirect` is to just drop them on the floor, so we'll have to figure out how to measure that when we add invalid cases.